pandas/io/parquet.py

```diff
@@ -90,10 +90,20 @@ def _get_path_or_handle(
     path_or_handle = stringify_path(path)
     if is_fsspec_url(path_or_handle) and fs is None:
         fsspec = import_optional_dependency("fsspec")
+        if storage_options is None:
+            pa = import_optional_dependency("pyarrow")
+            pa_fs = import_optional_dependency("pyarrow.fs")
 
-        fs, path_or_handle = fsspec.core.url_to_fs(
-            path_or_handle, **(storage_options or {})
-        )
+            try:
+                fs_arrow = import_optional_dependency("fsspec.implementations.arrow")
+                fs, path_or_handle = pa_fs.FileSystem.from_uri(path)
+                fs = fs_arrow.ArrowFSWrapper(fs)
+            except (TypeError, pa.ArrowInvalid):
+                pass
+        if fs is None:
+            fs, path_or_handle = fsspec.core.url_to_fs(
+                path_or_handle, **(storage_options or {})
+            )
     elif storage_options and (not is_url(path_or_handle) or mode != "rb"):
```

Comment:
Should we also check that no storage_options are passed if the user manually specified a pyarrow filesystem for `filesystem` ?

Reply:
Sure. Added validation that `filesystem` is a pyarrow or fsspec object too.